frameworks/kafka/tests/test_health_check.py

```diff
@@ -0,0 +1,139 @@
+import logging
+import re
+import pytest
+
+import sdk_auth
+import sdk_install
+import sdk_upgrade
+import sdk_utils
+
+from tests import auth
+from tests import config
+from security import transport_encryption
+
+from tests.health_check import check_health_check_logs
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope="module", autouse=True)
+def service_account(configure_security):
+    """
+    Sets up a service account for use with TLS.
+    """
+    try:
+        name = config.SERVICE_NAME
+        service_account_info = transport_encryption.setup_service_account(name)
+
+        yield service_account_info
+    finally:
+        transport_encryption.cleanup_service_account(config.SERVICE_NAME, service_account_info)
+
+
+@pytest.fixture(scope="module", autouse=True)
+def kerberos(configure_security):
+    try:
+        kerberos_env = sdk_auth.KerberosEnvironment()
+
+        principals = auth.get_service_principals(config.SERVICE_NAME, kerberos_env.get_realm(), health_check=True)
+        kerberos_env.add_principals(principals)
+        kerberos_env.finalize()
+
+        yield kerberos_env
+
+    finally:
+        kerberos_env.cleanup()
+
+
+@pytest.fixture(scope="module", autouse=True)
+def default_install():
+    sdk_install.uninstall(config.PACKAGE_NAME, config.SERVICE_NAME)
+    try:
+        sdk_install.install(
+            config.PACKAGE_NAME,
+            config.SERVICE_NAME,
+            config.DEFAULT_BROKER_COUNT,
+            timeout_seconds=30 * 60,
+        )
+        yield {**{"package_name": config.PACKAGE_NAME}}
+    finally:
+        sdk_install.uninstall(config.PACKAGE_NAME, config.SERVICE_NAME)
+
+
+@pytest.fixture(scope="module", params=["PORT", "FUNCTIONAL"])
+def health_check_method(request):
+    return request.param
+
+
+@pytest.fixture(scope="module", params=["DEFAULT", "SASL_PLAINTEXT", "SASL_SSL", "SSL"])
```

Comment:
you don't have to `parametrize` this?

Reply:
Pytest is parameterizing this during runtime. It is a fixture, I tried @pytest.parameterize but it worked on test functions not fixtures :man_shrugging: I guess this is "@pytest.parameterize" for fixtures.